src/controllers/controller.h

```diff
@@ -200,6 +206,39 @@ class ControllerJSProxy : public QObject {
         m_pController->send(data, data.length());
     }
 
+    // Available charsets should be available here:
+    // http://www.iana.org/assignments/character-sets/character-sets.xhtml
+    Q_INVOKABLE QJSValue convertEncoding(const QString& targetCharset, QString& value) {
+        std::shared_ptr<QJSEngine> pJsEngine = m_pController->getScriptEngine()->jsEngine();
+        VERIFY_OR_DEBUG_ASSERT(pJsEngine) {
+            return QJSValue::UndefinedValue;
+        }
+
+        icu::UnicodeString unicodeString;
+        UErrorCode errorCode;
+        UConverter* latin9Converter = ucnv_open(targetCharset.toLocal8Bit().data(), &errorCode);
+
+        if (!U_FAILURE(errorCode)) {
+            ucnv_close(latin9Converter);
+            return QJSValue::UndefinedValue;
+        }
+
+        char* result = nullptr;
+        ucnv_fromUChars(latin9Converter,
+                result,
+                0,
+                &value.data()->unicode(),
+                value.length(),
+                &errorCode);
```

Comment:
this is completely wrong judging by the documentation. https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/ucnv_8h.html#a4e7ebc80f4ee5a02472b139dbaa99c0f

Reply:
It feels like it, yes. `UChar` is an alias to `char16_t` but I can't figure out how to get a `char16_t*` from QString.

Reply:
I think you want `QString::utf16` for reading. Then write directly into an appropriately [pre-allocated QByteArray](https://doc.qt.io/qt-6/qbytearray.html#QByteArray-4)

Reply:
Wait… `QString::utf16()` returns `const ushort *` while `ucnv_fromUChars` expects `UChar*` as input and `char*` as output. How can that work? Can I just `(UChar*)QString::utf16()` and `(char*) myQByteArray`?

Reply:
kind of. I'd add some checks though. like `static_assert(sizeof(ushort) == sizeof(UChar) && std::is_signed_v<ushort> == std::is_signed_v<UChar> && std::is_convertible_v<ushort, UChar>)`. Not sure thats sufficient but good enough. For the output, you can probably do `reinterpret_cast<char *>(QByteArray::data())` (**assuming you have pre-allocated a large enough buffer**).